cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_dpc.cpp

```diff
@@ -17,6 +17,7 @@
 #include "oneapi/dal/backend/primitives/selection/kselect_by_rows.hpp"
 #include "oneapi/dal/backend/primitives/selection/kselect_by_rows_simd.hpp"
 #include "oneapi/dal/backend/primitives/selection/kselect_by_rows_quick.hpp"
+#include "oneapi/dal/backend/primitives/selection/kselect_by_rows_single_col.hpp"
```

Comment:
Are all of these files supposed to be called on the algorithm side? Or some of them just contain internal implementation of kselect? I am wondering how I should distinguish "public" API of kselect implementation from internal implementation. At now, the `selection` folder looks like a mess for me with a lot of files. It's also unclear why `row_partitioning.hpp` is in this folder too - the folder should contain only selection primitives I think.